### PR TITLE
Add Javascript tests to Travis CI. (#2093)

### PR DESCRIPTION
**Pull request by MatthewShao**
Adding our Javascript tests as a seperated matrix item to Travis CI, set `allow_failures` for now. 
The test result is look like this: [https://travis-ci.org/MatthewShao/mitmproxy/jobs/208819088](https://travis-ci.org/MatthewShao/mitmproxy/jobs/208819088)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,11 +5,17 @@ env:
   global:
     - CI_DEPS=codecov>=2.0.5
     - CI_COMMANDS=codecov
+    - MITMWEB_DIR=web
```
Comment (mhils) [inline .travis.yml]: Can we just hardcode this below please?

File: .travis.yml [hunk 2]
```diff
@@ -47,6 +53,15 @@ matrix:
       env: TOXENV=individual_coverage
     - python: 3.5
       env: TOXENV=docs
+    - language: node_js
+      node_js:
+          - "6"
+          - "iojs"
```
Comment (mhils) [inline .travis.yml]: Why not just "node" only?

File: .travis.yml [hunk 3]
```diff
@@ -47,6 +53,15 @@ matrix:
...
+      install: cd $MITMWEB_DIR && npm install
```
Comment (mhils) [inline .travis.yml]: Can we use yarn? I'd like to have a bit more consistency :) 
https://blog.travis-ci.com/2016-11-21-travis-ci-now-supports-yarn

**Comment by MatthewShao** [comment]
`yarn` is much faster than `npm install` :smiley:

**Comment by mhils** [comment]
Awesome, thanks!

**Pull request merged**